ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -422,52 +422,41 @@ public Node visitQuerySpecification(SqlBaseParser.QuerySpecificationContext cont
     return selectItems;
   }
 
-  private List<SelectItem> getSelectStartItems(final SelectItem selectItem, final Relation from) {
+  private List<SelectItem> getSelectStarItems(final SelectItem selectItem, final Relation from) {
     List<SelectItem> selectItems = new ArrayList<>();
     AllColumns allColumns = (AllColumns) selectItem;
 
+    final NodeLocation location = allColumns.getLocation().orElse(null);
     if (from instanceof Join) {
-      Join join = (Join) from;
-      AliasedRelation left = (AliasedRelation) join.getLeft();
-      StructuredDataSource
-          leftDataSource =
-          dataSourceExtractor.getMetaStore().getSource(left.getRelation().toString());
-      if (leftDataSource == null) {
-        throw new InvalidColumnReferenceException(left.getRelation().toString()
-                                                  + " does not exist.");
-      }
-      AliasedRelation right = (AliasedRelation) join.getRight();
-      StructuredDataSource rightDataSource =
-          dataSourceExtractor.getMetaStore().getSource(right.getRelation().toString());
-      if (rightDataSource == null) {
-        throw new InvalidColumnReferenceException(right.getRelation().toString()
-                                                  + " does not exist.");
-      }
-      for (Field field : leftDataSource.getSchema().fields()) {
-        QualifiedNameReference qualifiedNameReference =
-            new QualifiedNameReference(
-                allColumns.getLocation().get(),
-                QualifiedName.of(left.getAlias() + "." + field.name())
-            );
-        SingleColumn newSelectItem =
-            new SingleColumn(
-                qualifiedNameReference,
-                left.getAlias() + "_" + field.name()
-            );
-        selectItems.add(newSelectItem);
-      }
-      for (Field field : rightDataSource.getSchema().fields()) {
-        QualifiedNameReference qualifiedNameReference =
-            new QualifiedNameReference(
-                allColumns.getLocation().get(),
-                QualifiedName.of(right.getAlias() + "." + field.name())
-            );
-        SingleColumn newSelectItem =
-            new SingleColumn(
-                qualifiedNameReference,
-                right.getAlias() + "_" + field.name()
-            );
-        selectItems.add(newSelectItem);
+      final Join join = (Join) from;
+      if (allColumns.getPrefix().isPresent()) {
+        final String alias = allColumns.getPrefix().get().toString();
```

Comment:
Not sure I follow how the aliasing logic works. Is the alias here `tt` and `tt2` in `select a, b, c FROM test_stream tt left join test_stream_2 tt2`? If so, how is the right alias being handled? If not, is the alias `*` in `select tt.*, t1.f1 FROM ...`? I assume this branch is the bug fix but am not clear about how it is working.

Reply:
the alias is the `tt`, i.e., the alias that has been used for the table. If we had `tt.*` and `tt2.*` then this block of code would be called twice. So we'd get all columns with the correct aliasing. So for your second query it would get called with an `AllColumns` that has an alias of `tt`, so we add only the columns from the source it is related to. The `tt.f1` is a `SingleColumn` so gets added separately in the method that calls this.

Reply:
Great. thanks, I had missed the calling context.